coverage_point_calculator/src/lib.rs

```diff
@@ -0,0 +1,824 @@
+//!   - CBRS Radio's location is always trusted because of GPS.
+//!
+//! - [Speedtest]
+//!   - The latest 6 speedtests will be used.
+//!   - There must be more than 2 speedtests.
+//!
+//! - [CoveredHex]
+//!   - If a Radio is not [BoostedHexStatus::Eligible], boost values are removed before calculations.
+//!
+//! [modeled-coverage]:        https://github.com/helium/HIP/blob/main/0074-mobile-poc-modeled-coverage-rewards.md#outdoor-radios
+//! [provider-boosting]:       https://github.com/helium/HIP/blob/main/0084-service-provider-hex-boosting.md
+//! [wifi-aps]:                https://github.com/helium/HIP/blob/main/0093-addition-of-wifi-aps-to-mobile-subdao.md
+//! [qos-score]:               https://github.com/helium/HIP/blob/main/0098-mobile-subdao-quality-of-service-requirements.md
+//! [oracle-boosting]:         https://github.com/helium/HIP/blob/main/0103-oracle-hex-boosting.md
+//! [hex-limits]:              https://github.com/helium/HIP/blob/main/0105-modification-of-mobile-subdao-hex-limits.md
+//! [prevent-gaming]:          https://github.com/helium/HIP/blob/main/0107-preventing-gaming-within-the-mobile-network.md
+//! [cbrs-experimental]:       https://github.com/helium/HIP/blob/main/0113-reward-cbrs-as-experimental.md
+//! [mobile-poc-blog]:         https://docs.helium.com/mobile/proof-of-coverage
+//! [boosted-hex-restriction]: https://github.com/helium/oracles/pull/808
+//!
+pub use crate::{
+    hexes::CoveredHex,
+    location::LocationTrust,
+    speedtest::{BytesPs, Speedtest},
+};
+use coverage_map::SignalLevel;
+use rust_decimal::Decimal;
+use rust_decimal_macros::dec;
+
+mod hexes;
+mod location;
+mod speedtest;
+
+pub type Result<T = ()> = std::result::Result<T, Error>;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("signal level {0:?} not allowed for {1:?}")]
+    InvalidSignalLevel(SignalLevel, RadioType),
+}
+
+/// Output of calculating coverage points for a Radio.
+///
+/// The data in this struct may be different from the input data, but
+/// it contains the values used for calculating coverage points.
+///
+/// - If more than the allowed speedtests were provided, only the speedtests
+///   considered are included here.
+///
+/// - When a radio covers boosted hexes, [CoveragePoints::location_trust_scores] will contain a
+///   trust score _after_ the boosted hex restriction has been applied.
+///
+/// - When a radio is not eligible for boosted hex rewards, [CoveragePoints::covered_hexes] will
+///   have no boosted_multiplier values.
+#[derive(Debug, Clone)]
+pub struct CoveragePoints {
+    /// Total Rewards Shares earned by the Radio.
+    ///
+    /// Includes Coverage and Backhaul.
+    /// Hex Coverage points * location trust multiplier * speedtest trust multiplier
+    pub reward_shares: Decimal,
+    /// Total Points of Coverage for a Radio.
+    ///
+    /// Does not include Backhaul.
+    /// Hex coverage points * location trust multiplier
+    pub total_coverage_points: Decimal,
```

Comment:
given we have reward_shares and then the values for each individual param making up that value why is total_coverage_points required ?

Reply:
`reward_shares` is used for calculating poc_rewards. 
`total_coverage_points` is the coverage points (post location trust multiplier, pre speedtest multiplier) value that is put in the `RadioReward` proto. 

Both values are provided in the struct so whoever is using them only has to round the final value, and there's no worry about someone trying to derive their own value outside the calculator.